Write an article on DDEV Value Proposition (#54)
ddev/ddev.com: src/content/blog/why-support-ddev.md

pubDate: 2023-05-24
summary: "Explaining the value of DDEV that makes financial support of the project worth doing"
author: Randy Fay
featureImage:
    src: /img/blog/2023/05/developers-at-table.png
    alt: Image of a team of web developers working together at a table and delighting in the project they have in front of them
categories:
  - Community
---

The DDEV project is committed to setting itself to be sustainable for the long term. We know that your choice of a tool is a signficant investment of resources, and we want your choice to last and be rewarding for the long term. In order to ensure that, we're doing a fund drive, explaining to teams, organizations, and individuals why it's important to financially support the project. Our key priority right now is adding [another paid maintainer](recruiting-maintainers.md) and a documentation/marketing maintainer. We'll be be updating you regularly about this initiative.

## What Sets DDEV Apart

- **Installation**: DDEV is easy to install. It's a single binary backed by Docker providers everywhere.
- **Project Setup**: Adding DDEV to a project is as easy as `ddev config` and can take as little as a minute.

> - **Reliability**: DDEV's behavior is predictable and reliable. Automated testing on all supported platforms and manual testing of every new feature make it rock solid.

Reviewer: When i've first read that sentence 

> Automated testing on all supported platforms and manual testing of every new feature make it rock solid.

 i thought "make" refers to DDEV instead of the automated and manual tests. even though it is clear written correctly it is still somehow hard to read and process. but i haven't had an idea for an alternative suggestion yet.

> - **Reliability**: DDEV's behavior is predictable and reliable. Automated testing on all supported platforms and manual testing of every new feature make it rock solid.

Reviewer 2: Shorter and warmer:

```suggestion
- **Reliability**: DDEV's behavior is rock solid. We automate tests on our supported platforms and manually test every new feature.
```

> - **Multiple OS/Architecture Support**: Even if your team members use macOS (Apple Silicon or Intel), Windows WSL2, Linux, and traditional Windows you can expect consistent behavior across all these environments for the entire team.

Reviewer: i always try to avoid parenthesis like `(apple silicon or intel)`, and based on the title ddev is not only behaving consistently on these different platforms but is also available for each of them. not a perfect draft but how about something in the direction of: 

`DDEV is available for MacOS on Apple Silicon, MacOS on Apple Intel, Windows WSL2, regular Windows, and Linux. No matter which environment your team members use DDEV behaves consistently and makes collaboration a breeze. 
`

that way you cover that ddev is available on the different environments and express in the second sentence that the behavior is consistent and empowers teams with a heterogenous env setup to have a flawless collaboration across environments.

Reviewer 2: +1 on avoiding parenthetical statements that hinder flow! I use them constantly and have to re-edit.

> - **Multiple OS/Architecture Support**: Even if your team members use macOS (Apple Silicon or Intel), Windows WSL2, Linux, and traditional Windows you can expect consistent behavior across all these environments for the entire team.

Reviewer 2: I know the Mac and Windows architecture are critically-important DDEV features, but they get in the way of making this point which is about the team’s platform diversity _not_ being a barrier:

```suggestion
- **Multiple OS/Architecture Support**: Consistent behavior for the entire team across macOS, Windows, and Linux.
```

If it’s important to demonstrate depth of commitment to those platforms, maybe make that its own “DDEV runs on that” bullet point where you can address the nuances of each platform and link to the post where you ran DDEV on a Raspberry Pi.

> - **Team Consistency**: Teams typically check in the DDEV configuration, meaning that team members can check out the project and `ddev start` and they'll have everything running right away in less than a minute.

Reviewer: check in and check out probably refers to version control. would it make sense to make that more explicit? and maybe also make the consequence more clear? that each team member is using the same configuration and in consequence the same dev setup that way?

> - **Team Consistency**: Teams typically check in the DDEV configuration, meaning that team members can check out the project and `ddev start` and they'll have everything running right away in less than a minute.

Reviewer 2: Don’t assume your reader knows “check in” is Git, but explain the concept and benefit anyway:

```suggestion
- **Team Consistency**: Teams can share DDEV configuration so that anyone can run `ddev start` and have the same setup quickly on their machine.
```

> - **Web server and Database Support**: Out of the box, DDEV supports Nginx and Apache webservers, and directly supports MariaDB, MySQL, and PostgreSQL databases.

Reviewer: Shouldn't `Server` in `Web Server` be upper case? and the term webserver should be consistent. Either webserver or web server. and i would suggest to strike "directly support". an alternative might be something like: 

`Out of the box, DDEV provides support for Nginx and Apache web servers, as well as MariaDB, MySQL, and PostgreSQL databases.`

> - **Web server and Database Support**: Out of the box, DDEV supports Nginx and Apache webservers, and directly supports MariaDB, MySQL, and PostgreSQL databases.

Reviewer 2: Attempt to make this a bit more accessible to someone that doesn’t work with web servers and databases:

```suggestion
- **Web server and Database Support**: DDEV directly supports commonly-used software like nginx and Apache web servers, and MariaDB, MySQL, and PostgreSQL databases.
```

> - **Add-On Ecosystem**: A huge number of non-core features are supported via add-ons, including Apache Solr, Elasticsearch, and dozens of others. There are officially supported add-ons and a large number contributed and maintained by the community.

Reviewer: how about the following, slightly modified, version: 

`A huge number of non-core features are supported via officially supported add-ons, including Apache Solr, Elasticsearch, and dozens of others. Aside the official add-ons there is a large number contributed add-ons maintained by the community.`

> - **Add-On Ecosystem**: A huge number of non-core features are supported via add-ons, including Apache Solr, Elasticsearch, and dozens of others. There are officially supported add-ons and a large number contributed and maintained by the community.

Reviewer 2: “Huge” and “large” are relative terms that don’t offer much—better to link to the best publicly-available list (which I think is in the docs?) and help the reader decide if they need to investigate. Shorten and link:

```suggestion
- **Add-On Ecosystem**: More features are available as first-party and community-contributed [add-ons](https://ddev.readthedocs.io/en/stable/users/extend/additional-services/), including Apache Solr and Elasticsearch.
```

I deliberately chose “first-party” instead of “officially supported” to indicate authorship/maintainership and avoid making it sound like “supported” vs. “unsupported”.

> - **Community**: DDEV's friendly and active community meets in Discord, the GitHub issue queue, and Stack Overflow. People from many different CMS and framework communities have years of experience and help each other.

Reviewer 2: “Meet” reads to me like “congregate” here, so the idea of people “meeting” in GitHub Issues and Stack Overflow seems a little strange.

More importantly though, I like the point about diversity of CMS/platform experience.

This might be a good place to link to a code of conduct or anything that reassures an outsider that it’s a welcoming and non-toxic community—even if subtly.

> - **Support**: DDEV's support is world-class and very active. Questions in the issue queue are often answered in minutes. Problems in Discord are often discussed and resolved very, very quickly. Bugs get early attention.

Reviewer: instead of `quickly` how about using `timely`? instead of `very, very quickly` simply `in a timely manner.`. And i wonder what does the sentence with `early attention` at the end mean, and does it add additional information? because it was already communicated in the previous sentences that the responsiveness in all the channels is high.

> - **Support**: DDEV's support is world-class and very active. Questions in the issue queue are often answered in minutes. Problems in Discord are often discussed and resolved very, very quickly. Bugs get early attention.

Reviewer 2: I bristle when marketers describe their own products or services as “world-class” because it’s an impressive-sounding business phrase that rings hollow. But I’m more curmudgeonly than most.

I’d earnestly describe what DDEV’s support is like instead, use “GitHub Issues” rather than the more jargony “issue queue,” and tighten everything up:

```suggestion
- **Support**: DDEV's support is swift, friendly, and competent. Discord questions and GitHub Issues are always answered, often in minutes. Bugs get early attention.
```

My usual experience with modern support is a mixture of non-response, lifelessness, and lack of care or expertise—hence the words I chose here which I feel are refreshingly and honestly accurate.

> - **Maintenance**: Careful maintenance means that new upstream capabilities (PHP versions, other tool changes) and upstream bugs get resolved and tested soon. Extensive automated testing means that upstream problems are detected quickly.

Reviewer: the title is `maintenance` and the sentence is started with `careful maintenance`. as a naive reader i ask myself is `maintenance` and `careful maintenance` something different? and the sentence is sort of hard to read. `upstream capabilities` does that mean `upstream features`? and the info in parenthesis is sort of confusing in the context of upstream capabilities. is a php version or other tool changes capability/feature?  plus the parenthesis make the sentence hard to read and process. and `extensive automated testing` sounds like ddev would test the code upsteam?

> - **Maintenance**: Careful maintenance means that new upstream capabilities (PHP versions, other tool changes) and upstream bugs get resolved and tested soon. Extensive automated testing means that upstream problems are detected quickly.

Reviewer 2: I’d drop this bullet point because you already conveyed the sense of diligence and care with testing and platform support. Feels like you’re calling attention to something you already told me.

> - **Open Source**: DDEV is completely Free and Open Source Software (FOSS). It also can work with FOSS Docker providers on every platform, so you're not tied to closed-source commercial solutions like Docker Desktop, although Docker Desktop is fully supported.

Reviewer: in the second sentence i would strike the `although docker desktop is supported` part. that is already mentioned in the beginning on the installation point (at least in my suggestion there ;) ) and maybe split the sentence into two?

`It works with available FOSS Docker providers on every platform. Thereby you are not tied to closed-source commercial solutions like Docker Desktop.`

> - **Open Source**: DDEV is completely Free and Open Source Software (FOSS). It also can work with FOSS Docker providers on every platform, so you're not tied to closed-source commercial solutions like Docker Desktop, although Docker Desktop is fully supported.

Reviewer 2: I’d drop the “it also” and Docker Desktop mention. It feels like a dig against Docker Desktop, which a lot of people are probably using? Seems like you can make the point about not being constrained without inviting needless scrutiny of Docker Desktop that may be more of a distraction.

```suggestion
- **Open Source**: DDEV is completely Free and Open Source Software (FOSS). It can work with FOSS Docker providers on every platform, so you're not tied to closed-source commercial solutions.
```

As written, it reads like “With DDEV you’re not stuck with closed-source things like the one you’re probably using, which we also support.”

> **We invite you to calculate the value DDEV brings to your work and consider making a financial contribution to support the project.** DDEV consistently delivers on the promise of making local development work for your projects, every time. How much time, energy, and resources do you save by using DDEV as opposed to maintaining your own solution or using an alternate? What would it cost to achieve the same efficiency and consistency without DDEV? Please calculate that value for yourself as you're considering how to financially support the project.

Reviewer: the only nitpick is the following sentence. 

> DDEV consistently delivers on the promise of making local development work for your projects, every time.

when i start reading i always think the "local development work"..."for your project" instead of "making local development".. "work for your project.

> **We invite you to calculate the value DDEV brings to your work and consider making a financial contribution to support the project.** DDEV consistently delivers on the promise of making local development work for your projects, every time. How much time, energy, and resources do you save by using DDEV as opposed to maintaining your own solution or using an alternate? What would it cost to achieve the same efficiency and consistency without DDEV? Please calculate that value for yourself as you're considering how to financially support the project.

Reviewer 2: As I wrote, I think that as a CTA this is phoning it in and asking the reader to convince themselves that DDEV’s worth their investment without offering much help. Even taking one example of audience member and demonstrating some napkin math would go a long way.

There is no “as you’re considering how to financially support the project” if they’re not convinced they _should_ in the first place.

Who is the target audience and what barriers do they have for local development and financial support of the project?

> To explore DDEV's strengths in more technical detail, see [What's so different about DDEV](whats-so-different-about-ddev-local.md).

Reviewer 2: Fix title and link:

```suggestion
To explore DDEV's strengths in more technical detail, see [What’s so different about DDEV?](/blog/whats-so-different-about-ddev-local/).
```
